checkstyle-tester/diff.groovy

```diff
@@ -373,6 +385,46 @@ def cloneRepository(repoName, repoType, repoUrl, commitId, srcDir) {
     println "$repoName is synchronized"
 }
 
+def getCloneCmd(repoType, repoUrl, srcDestinationDir) {
+    final String cloneCmd = ''
+    switch (repoType) {
+        case 'git':
+            cloneCmd = "git clone $repoUrl $srcDestinationDir"
+            break
+        default:
+            throw new IllegalArgumentException("Error! Unknown $repoType repository.")
+    }
+    return cloneCmd
+}
+
+def getCloneShallowCmd(repoType, repoUrl, srcDestinationDir, commitId) {
+    if ('git'.equals(repoType)) {
+        return "git clone --depth 1 --branch $commitId $repoUrl $srcDestinationDir"
+    }
+        throw new IllegalArgumentException("Error! Unknown repository type: $repoType")
+}
+
+def fetchAdditionalData(repoType, srcDestinationDir, commitId) {
+    def fetchCmd = ''
```

Comment:
Why couldn't we use `String` here?

Reply:
I personally think that best form of groovy is pure java code. In this case script is naturally growing and become and application by simple conversion.

Reply:
I disagree, but this is how we currently operate so let's go with it.

Reply:
Done.